### ISSUE TITLE
ez_sql_mysql escape() function spoils strings with the slashes

### ISSUE THREAD

**Issue by AndreyRa** [ezSQL/ezSQL#85]
I'm trying to INSERT json-encoded array in the TEXT field with ez_sql_mysql wrapper. I'm using escape() function for every string variable before it passes to the query. My string contains Unicode characters, so my array encoded like this: ["<foo>","'bar'","\"baz\"","&blong&","\u00e9"].

Now I found that all my slashes is missing.

I have searched through the code and found root cause of problems: wrong call of stripslahes() function at 177 line of the ez_sql_mysql.php:
return mysql_real_escape_string(stripslahes($str));

I have deleted stripslashes() call and now all works fine!

**Comment by gnowxilef** [comment]
feel free to submit a PR. :)

**Comment by bdewater** [comment]
As far as I can tell, the comment on [line 176 in ez_sql_mysqli.php](https://github.com/ezSQL/ezSQL/blob/d608f4c25f1c42b96de2b25cac983e6af8913b1e/mysqli/ez_sql_mysqli.php#L176) (and in other similar files) is incorrect because it always calls stripslashes, without checking if `magic_quotes_gpc` is enabled or not. In my case this corrupted image files I needed to store in a MySQL longblob column on a PHP 5.3 environment with magic quotes off. According to http://php.net/manual/en/security.magicquotes.php it has been removed as off PHP5.4, so this bug will cause more trouble in the future as installations are upgraded.

Afaik something like the code below (adapted from [this example](http://php.net/manual/en/function.get-magic-quotes-gpc.php) from the PHP manual) should replace line 188 in order to have it function like the comment describes:
```php
if ( get_magic_quotes_gpc() ) {
  $str = stripslashes($str);
}

return $this->dbh->escape_string($str);
```
